tests/booted/test-logically-bound.nu

```diff
@@ -0,0 +1,151 @@
+    }
+
+    for $container in $bound_containers {
+        let found = $image_names | where Names == [$container.image]
+        assert (($found | length) > 0) $"($container.image) not found"
+    }
+}
+
+def first_boot [] {
+    tap begin "bootc switch with bound images"
+
+    initial_setup
+
+    # build a bootc image that includes bound images
+    let images = echo [
+        { "bound": true, "image": "registry.access.redhat.com/ubi9/ubi-minimal:9.4", "name": "ubi-minimal" },
+        { "bound": false, "image": "quay.io/centos-bootc/centos-bootc:stream9", "name": "centos-bootc" }
+    ]
+
+    let containers = echo [{
+        "bound": true, "image": "docker.io/library/alpine:latest", "name": "alpine" 
+    }]
+
+    let image_name = "localhost/bootc-bound"
+    build_image $image_name $images $containers
+    bootc switch --transport containers-storage $image_name
+    verify_images $images $containers
+    tmt-reboot
+}
+
+def second_boot [] {
+    print "verifying second boot after switch"
+    assert equal $booted.image.transport containers-storage
+    assert equal $booted.image.image localhost/bootc-bound
+
+    # verify images are still there after boot
+    let images = echo [
+        { "bound": true, "image": "registry.access.redhat.com/ubi9/ubi-minimal:9.4", "name": "ubi-minimal" },
+        { "bound": false, "image": "quay.io/centos-bootc/centos-bootc:stream9", "name": "centos-bootc" }
+    ]
+
+    let containers = echo [{
+        "bound": true, "image": "docker.io/library/alpine:latest", "name": "alpine" 
+    }]
+    verify_images $images $containers
+
+    # build a new bootc image with an additional bound image
+    print "bootc upgrade with another bound image"
+    let image_name = "localhost/bootc-bound"
+    let more_images = $images | append [{ "bound": true, "image": "registry.access.redhat.com/ubi9/ubi-minimal:9.3", "name": "ubi-minimal-9-3" }]
+    build_image $image_name $more_images $containers
+    bootc upgrade
+    verify_images $more_images $containers
+    tmt-reboot
+}
+
+def third_boot [] {
+    print "verifying third boot after upgrade"
+    assert equal $booted.image.transport containers-storage
+    assert equal $booted.image.image localhost/bootc-bound
+
+    let images = echo [
+        { "bound": true, "image": "registry.access.redhat.com/ubi9/ubi-minimal:9.4", "name": "ubi-minimal" },
+        { "bound": true, "image": "registry.access.redhat.com/ubi9/ubi-minimal:9.3", "name": "ubi-minimal-9-3" },
+        { "bound": false, "image": "quay.io/centos-bootc/centos-bootc:stream9", "name": "centos-bootc" }
+    ]
+
+    let containers = echo [{
+        "bound": true, "image": "docker.io/library/alpine:latest", "name": "alpine" 
+    }]
+
+    verify_images $images $containers
+    tap ok
+}
+
+def main [] {
+    match $env.TMT_REBOOT_COUNT? {
+        null | "0" => first_boot,
+        null | "1" => second_boot,
+        null | "2" => third_boot,
```

Comment:
The last two `null` here are unreachable (only the first can match), a bit surprising nu doesn't error on this. I've updated the code to drop them.